system/jlib/jlzw.cpp

```diff
@@ -315,6 +315,20 @@ void CLZWCompressor::open(void *buf,size32_t max)
 
 
 size32_t CLZWCompressor::write(const void *buf,size32_t buflen)
+{
+    return doWrite((size32_t)-1, buf, buflen);
+}
+
+size32_t CLZWCompressor::limitWrite(size32_t limit, const void *buf,size32_t buflen)
```

Comment:
This may not be the best way to implement this.  Better may be to dynamically call something to adjust the limit before calling write.  That would avoid the complication of maxlen potentially changing in doWrite.  Something to discuss and revisit.